### PR TITLE
Update asset versioning and caching guide with new staleness ontology

### PR DESCRIPTION
**Pull request by sryza**
branch-name: asset-versioning-and-caching-update

### CODE REVIEW

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx [hunk 1]
```diff
@@ -226,24 +234,26 @@ def multiplied_number(versioned_number):
...
 width={2662}
-height={1748}
+height={1618}
 />
 
-Notice that `multiplied_number` is no longer stale, even though we didn't materialize it! Here's what happened: `versioned_number` is no longer stale, so `multiplied_number` is no longer automatically stale. Dagster then compared the data version of `versioned_number` last used to materialize `multiplied_number` to the current data version of `versioned_number`. Since this comparison shows that the data version of `versioned_number` hasn't changed, Dagster knows that `multiplied_number` isn't stale.
+Notice that `multiplied_number` is no longer has a label, even though we didn't materialize it! Here's what happened: the new materialization of `versioned_number` with the explicitly-supplied data version supercedes the code version of `versioned_number`. Dagster then compared the data version of `versioned_number` last used to materialize `multiplied_number` to the current data version of `versioned_number`. Since this comparison shows that the data version of `versioned_number` hasn't changed, Dagster knows that the change to the code version of `versioned_number` doesn't affecet `multiplied_number`.
```
Comment (schrockn) [inline docs/content/guides/dagster/asset-versioning-and-caching.mdx:250]: s/affecet/affect

**Review by schrockn — COMMENTED** [review]
I think this is pretty great.

Definitely the thing that sticks out is "Propagate Changes" as a weird phrasing given how everything else feels. I still wish we could do is "Materialize" instead of "Propagate Changes" and "Force Materialize" instead of what is currently "Materialize" for asset graphs that do code or data versioning.

This document is purely a reflection of this previous discussion, so I'm not proposing having the discussion here, but just noting that it is the only thing that sticks out in an otherwise excellent improvement.

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx (hunk 1 above)
Comment (schrockn) [inline docs/content/guides/dagster/asset-versioning-and-caching.mdx:250]: s/explicitly-supplied/explicitly supplied/

adverb + adjective not hyphenated

**Review by schrockn — APPROVED** [review]
Good stuff. I think the new verbiage lands well.

File: docs/content/guides/dagster/asset-versioning-and-caching.mdx (hunk 1 above)
Comment (smackesey) [reply docs/content/guides/dagster/asset-versioning-and-caching.mdx:250]: "is no longer has a label"

**Pull request merged**